cpp/src/parquet/schema_test.cc

```diff
@@ -767,35 +809,38 @@ static std::string Print(const NodePtr& node) {
 TEST(TestSchemaPrinter, Examples) {
   // Test schema 1
   NodeVector fields;
-  fields.push_back(Int32("a", Repetition::REQUIRED));
+  fields.push_back(Int32("a", Repetition::REQUIRED, 1));
 
   // 3-level list encoding
-  NodePtr item1 = Int64("item1");
-  NodePtr item2 = Boolean("item2", Repetition::REQUIRED);
+  NodePtr item1 = Int64("item1", Repetition::OPTIONAL, 4);
+  NodePtr item2 = Boolean("item2", Repetition::REQUIRED, 5);
   NodePtr list(
-      GroupNode::Make("b", Repetition::REPEATED, {item1, item2}, ConvertedType::LIST));
-  NodePtr bag(GroupNode::Make("bag", Repetition::OPTIONAL, {list}));
+      GroupNode::Make("b", Repetition::REPEATED, {item1, item2}, ConvertedType::LIST, 3));
+  NodePtr bag(
+      GroupNode::Make("bag", Repetition::OPTIONAL, {list}, /*logical_type=*/nullptr, 2));
   fields.push_back(bag);
 
   fields.push_back(PrimitiveNode::Make("c", Repetition::REQUIRED, Type::INT32,
-                                       ConvertedType::DECIMAL, -1, 3, 2));
+                                       ConvertedType::DECIMAL, -1, 3, 2, 6));
 
   fields.push_back(PrimitiveNode::Make("d", Repetition::REQUIRED,
-                                       DecimalLogicalType::Make(10, 5), Type::INT64, -1));
+                                       DecimalLogicalType::Make(10, 5), Type::INT64,
+                                       /*length=*/-1, 7));
 
-  NodePtr schema = GroupNode::Make("schema", Repetition::REPEATED, fields);
+  NodePtr schema = GroupNode::Make("schema", Repetition::REPEATED, fields,
+                                   /*logical_type=*/nullptr, 0);
 
   std::string result = Print(schema);
-  std::string expected = R"(message schema {
-  required int32 a;
-  optional group bag {
-    repeated group b (List) {
-      optional int64 item1;
-      required boolean item2;
+  std::string expected = R"(repeated group field_id=0 schema {
```

Comment:
Just so I understand, the fact that the field ids are DFS-ordered is a test decision, not a Parquet requirement, right?

Reply:
Right, that's how they're auto-assigned by the library but another library may set different ids